### PR TITLE
Merging dev into main

### PR DESCRIPTION
**Pull request by twarin**
Added/tweaked some endpoints for stories, authentication, and permissions.
Added title and is_featured to story model.

### CODE REVIEW

File: config/settings.py [hunk 1]
```diff
@@ -151,15 +156,36 @@
 
 REST_USE_JWT = True
 SIMPLE_JWT = {
-    "ACCESS_TOKEN_LIFETIME": timedelta(days=1),
-    "REFRESH_TOKEN_LIFETIME": timedelta(days=10),
-    "ROTATE_REFRESH_TOKENS": False,
-    "BLACKLIST_AFTER_ROTATION": True,
+    'ACCESS_TOKEN_LIFETIME': timedelta(minutes=180),
```
Comment (BrawnaNG) [inline config/settings.py:159]: Why so much shorter?

File: config/urls.py [hunk 2]
```diff
@@ -15,20 +15,22 @@
...
     path("api/v1/category", include("category.urls", namespace="category")),
     path("api/v1/comment", include("comment.urls", namespace="comment")),
     path("api/v1/story", include("story.urls", namespace="story")),
     path("api/v1/tag", include("tag.urls", namespace="tag")),
-    path("api/token/", TokenObtainPairView.as_view(), name="token_obtain_pair"),
-    path("api/token/refresh/", TokenRefreshView.as_view(), name="token_refresh"),
-    path("api/token/verify/", TokenVerifyView.as_view(), name="token_verify"),
+    path("api/v1/token/", MyTokenObtainPairView.as_view(), name="token_obtain_pair"),
+    path("api/v1/token/refresh/", TokenRefreshView.as_view(), name="token_refresh"),
+    path("api/v1/token/verify/", TokenVerifyView.as_view(), name="token_verify"),
```
Comment (BrawnaNG) [inline config/urls.py:35]: Good catch! Dunno if we'll ever do a v2, but having this consistency is great.

**Review by BrawnaNG — APPROVED** [review]
Looks good! Two minor points, but merging.

File: config/settings.py (hunk 1 above)
Comment (twarin) [reply config/settings.py:159]: I was going by some sample code found online. The idea is that the shorter access token lifetime gives less scope for abuse. It's also easier to test. But honestly I don't know if we need to do it this way.

**Pull request merged**